### ISSUE TITLE
[Feature]: "Freeze" mode for testing out some changes and discarding all changes instantly

### ISSUE THREAD

**Issue by gituser1000000** [AndroidIDEOfficial/AndroidIDE#1262]
### Feature description

Sometimes we need to test out some changes and see how things would work out. If we don't like the results, we could discard everything and get back to the state before testing.

The current ways of doing (or trying to do) this are:

1. Make a copy of the project and copy it back after testing.

2. Use the undo/redo history and keep pressing undo repeatedly for each file until we find the old history states. We may also hit a limit on how far we can go back.

Needless to say, these two methods don't seem to do the job well enough, especially if being "instant" is a requirement.

### What version of AndroidIDE you're using?

v2.5.2 (debug builds)

### Duplicate issues

- [X] This issue has not been reported yet.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by itsaky** [comment]
I don't think this should be a feature in the IDE. `git` provides the exact same features with commits, branches, stashes, etc.

> Sometimes we need to test out some changes and see how things would work out. If we don't like the results, we could discard everything and get back to the state before testing.

With `git` :
- Commit your current changes
- Make the changes you want to test out
- If the changes don't work out, revert/reset to the previous commit.

I'm closing the issue as I don't think this feature is necessary. Reopen the issue if you think I'm mistaken.

**Comment by gituser1000000** [comment]
I was thinking of having this useful feature on the UI via a single quick-and-simple freeze/unfreeze button which could be used by all users of any level and not just by users who know how to or enjoy opening up a terminal to execute commands.

I am going to keep this closed and leave it up to you to consider in the future when things aren't so busy.

**Comment by gituser1000000** [comment]
Just as a reference for future readers, I'm going state some differences between the proposed feature and the one in the reason for not being planned.

freeze/unfreeze UI button:
-For simple smaller changes (likely not for keeping) that we just want to build and see the results right away
git:
-For bigger changes that most likely are going to be worked on for a while

freeze/unfreeze UI button:
-Only requires preserving the files and putting them back, nothing else;
git:
-Overkill with all the unused abilities not needed for the suggested feature;

freeze/unfreeze UI button:
-instant access from the UI that shouldn't even break the flow of typing your current statement or losing train of thought
git:
-certainly not instant and needs to open up a terminal to execute commands, which includes typing in the project name

The bottom line is this feature would provide an instant and seamless way to quickly test out some changes on-the-fly. It could simply be thought of as an undo on steroid, or an undo with batched changes, that works over multiple files.

**Comment by itsaky** [comment]
Still, I don't think this feature is necessary (git is still the best option in my opinion). But if we do have more requests to add this, we'll consider working on it.
